cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -184,69 +194,145 @@ class ndview : public ndarray_base<axis_count, order> {
+    ///
+    /// @param data    The array that stores a homogeneous data block.
+    /// @param shape   The shape of the created multidimensional view.
+    ///
+    /// @return The new multidimensional view instance.
     static ndview wrap_mutable(const array<T>& data, const shape_t& shape) {
         ONEDAL_ASSERT(data.has_mutable_data());
         ONEDAL_ASSERT(data.get_count() >= shape.get_count());
         return wrap(data.get_mutable_data(), shape);
     }
 
+    /// Creates a new strided multidimensional view instance from a mutable array.
+    ///
+    /// @param data    The array that stores a homogeneous data block.
+    /// @param shape   The shape of the created multidimensional view.
+    /// @param strides The strides of the created multidimensional view.
+    ///
+    /// @return The new multidimensional view instance.
     static ndview wrap_mutable(const array<T>& data, const shape_t& shape, const shape_t& strides) {
         ONEDAL_ASSERT(data.has_mutable_data());
         ONEDAL_ASSERT(data.get_count() >= shape.get_count());
         return wrap(data.get_mutable_data(), shape, strides);
     }
 
+    /// Creates a new 1d view instance from an immutable array.
+    ///
+    /// @param data    The array that stores a homogeneous data block.
+    ///
+    /// @return The new 1d view instance.
     template <std::int64_t d = axis_count, typename = std::enable_if_t<d == 1>>
     static ndview wrap(const array<T>& data) {
         return wrap(data.get_data(), { data.get_count() });
     }
 
+    /// Creates a new 1d view instance from a mutable array.
+    ///
+    /// @param data    The array that stores a homogeneous data block.
+    ///
+    /// @return The new 1d view instance.
     template <std::int64_t d = axis_count, typename = std::enable_if_t<d == 1>>
     static ndview wrap_mutable(const array<T>& data) {
         ONEDAL_ASSERT(data.has_mutable_data());
         return wrap(data.get_mutable_data(), { data.get_count() });
     }
 
+    /// The pointer to the memory block holding immutable data.
     const T* get_data() const {
         return data_;
     }
 
+    /// The pointer to the memory block holding mutable data.
     T* get_mutable_data() const {
         ONEDAL_ASSERT(data_is_mutable_);
         return const_cast<T*>(data_);
     }
 
+    /// Returns whether the view contains data or not.
     bool has_data() const {
         return this->get_count() > 0;
     }
 
+    /// Returns whether the view contains mutable data or not.
     bool has_mutable_data() const {
         return this->has_data() && this->data_is_mutable_;
     }
 
+    /// Returns a reference to the element of 1d immutable view at specified location.
+    ///
+    /// @note This method does not perform boundary checks in release build.
+    ///       In debug build, the method asserts that the index is out of the bounds.
+    ///
+    /// @note Should be used carefully in performance-critical parts of the code
```

Comment:
nit: The fact that the method is in the header means that it is implicitly inline. It is likely to be inlined, but not guaranteed to be. The comment should reflect this

```
@note Should be used carefully in performance-critical parts of the code, as inlining is not guaranteed.
```